src/commands/upgrade.rs

```diff
@@ -0,0 +1,181 @@
+use crate::{load_metadata, Config, BINDINGS_CRATE_NAME};
+use anyhow::{Context, Result};
+use cargo_component_core::command::CommonOptions;
+use cargo_metadata::Metadata;
+use clap::Args;
+use std::{fs, os::unix::process::CommandExt, path::PathBuf};
```

Comment:
_Sigh_, I didn't even notice myself adding these. And I even thought to myself "huh, that's funny, I don't remember Windows supporting exec, but I guess I've been away for a while..." 😅